src/Interpreters/ClusterFunctionReadTask.cpp

```diff
@@ -29,7 +30,8 @@ ClusterFunctionReadTaskResponse::ClusterFunctionReadTaskResponse(ObjectInfoPtr o
     if (object->data_lake_metadata.has_value())
         data_lake_metadata = object->data_lake_metadata.value();
 
-    file_meta_info = object->file_meta_info;
+    if (context->getSettingsRef()[Setting::allow_experimental_iceberg_read_optimization])
```

Comment:
It feels a bit weird that `file_meta_info` is created conditionally, if the goal is to to send (or rather NOT send it), modifying `ClusterFunctionReadTaskResponse::serialize` seems to be more reasonable.